x/valset/keeper/keeper.go

```diff
@@ -525,14 +549,21 @@ func (k Keeper) GetSigningKey(ctx sdk.Context, valAddr sdk.ValAddress, chainType
 }
 
 // IsJailed returns if the current validator is jailed or not.
-func (k Keeper) IsJailed(ctx sdk.Context, val sdk.ValAddress) bool {
-	return k.staking.Validator(ctx, val).IsJailed()
+func (k Keeper) IsJailed(ctx context.Context, val sdk.ValAddress) (bool, error) {
+	a, err := k.staking.Validator(ctx, val)
+	if err != nil {
+		return a.IsJailed(), err
+	}
+	return a.IsJailed(), nil
 }
 
-func (k Keeper) Jail(ctx sdk.Context, valAddr sdk.ValAddress, reason string) error {
+func (k Keeper) Jail(_ctx context.Context, valAddr sdk.ValAddress, reason string) error {
+	ctx := sdk.UnwrapSDKContext(_ctx)
 	jailTime := ctx.BlockTime()
-	val := k.staking.Validator(ctx, valAddr)
-
+	val, err := k.staking.Validator(ctx, valAddr)
+	if err != nil {
+		return whoops.Errorf(err.Error()).Format()
```

Comment:
Can you explain what exactly you're doing here - and why?

Reply:
If i return the `error` ,i'm getting an error from the integration test. So i have used this error
And all the below return use the `whoops.Errof`

Reply:
We're trying to remove this `whoops` library. But it's out of scope for this. Thanks for clearing it up.

Reply:
I have returned the actual error now